cmake-ports.cmake

```
@@ -238,7 +248,7 @@ function(declare_port specifier result)
 
   if(ARGV_MESON)
     configure_meson_port()
-  elseif(ARGV_AUTOTOOLS)
+  elseif(ARGV_AUTOTOOLS OR ARGV_ENTRYPOINT)
```

```suggestion
  elseif(ARGV_AUTOTOOLS)
```

This way we can allow `ENTRYPOINT` to mean different things for different port types and not tie it implicitly to Autotools.

---

So the idea, is to have two differents macros?

I started with that approach but as two macros were almost the same, this is why I decided to refacto.

But yeah it is better to have a way to let both sides evolve on their own in the future.

---

It won't be a separate macro, no. `ENTRYPOINT` should be just another build system argument, like `ARGS`, and not tied to any one particular build system.

---

I am not sure to follow … 
Autotools is about to call configure. Here we want to call another script instead.

---

Oh yeah you want to keep the same mechanic but just remove the arg check here.

---

Yep, so you'd need to pass `AUTOTOOLS` to signal "this is an Autotools build" and `ENTRYPOINT` to signal "use this entrypoint instead of the usual `./configure`".

It's not inconceivable that the combination of e.g. `CMAKE` and `ENTRYPOINT` could mean "use this top-level `CMakeLists.txt` instead".